packages/react/src/components/Authenticator/FederatedSignIn/FederatedSignInButtons/__tests__/FederatedSignInButton.test.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+
+import { FederatedIdentityProviders } from '@aws-amplify/ui';
+import * as ReactCoreModule from '@aws-amplify/ui-react-core';
+
+import { FederatedSignInButton } from '../FederatedSignInButton';
+
+const SUPPORTED_PROVIDERS = [
+  FederatedIdentityProviders.Amazon,
+  FederatedIdentityProviders.Apple,
+  FederatedIdentityProviders.Facebook,
+  FederatedIdentityProviders.Google,
+];
+
+const useAuthenticatorMock = {
+  toFederatedSignIn: jest.fn(),
+};
+
+jest
+  .spyOn(ReactCoreModule, 'useAuthenticator')
+  .mockImplementation(() => useAuthenticatorMock as any);
```

Comment:
Can we use another type other than `any`? Otherwise this strips all type information we may want to keep.

```suggestion
  .mockImplementation(() => useAuthenticatorMock as any);
```

Reply:
This is casted to `UseAuthenticator` now: [0ceada4](https://github.com/aws-amplify/amplify-ui/pull/2706/commits/0ceada4f4011a471f52633dc9a6e74d7521a679e)